pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/BaseClusterIntegrationTestSet.java

```diff
@@ -320,19 +320,17 @@ public void testQueriesFromQueryFile()
     assertNotNull(resourceUrl);
     File queryFile = new File(resourceUrl.getFile());
 
-    int maxNumQueriesToSkipInQueryFile = getMaxNumQueriesToSkipInQueryFile();
     try (BufferedReader reader = new BufferedReader(new FileReader(queryFile))) {
       while (true) {
-        int numQueriesSkipped = RANDOM.nextInt(maxNumQueriesToSkipInQueryFile);
-        for (int i = 0; i < numQueriesSkipped; i++) {
-          reader.readLine();
-        }
-
         String queryString = reader.readLine();
         // Reach end of file.
         if (queryString == null) {
           return;
         }
+        // Skip commented line and empty line.
+        if (queryString.startsWith("#") || queryString.isEmpty()) {
```

Comment:
Use `queryString.trim()` to include leading whitespaces before # or empty lines with only whitespaces.